### PR TITLE
tikv-server: compatible with pd that did not implement configpb.Config service

### PR DESCRIPTION
**Pull request by NingLin-P**
###  What have you changed?

If tikv server connect to pd that did not implement configpb.Config service, print warn log and disable dynamic config.

###  What is the type of the changes?

- Bugfix (a change which fixes an issue)

###  How is the PR tested?

- Manual test (add detailed scripts or steps below)

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
No

###  Does this PR affect `tidb-ansible`?
Nol)

### CODE REVIEW

File: cmd/src/server.rs [hunk 1]
```diff
@@ -234,6 +215,47 @@ impl TiKVServer {
...
+            Ok((v, mut cfg)) => {
+                cfg.server.advertise_addr = advertise_addr;
+                cfg.server.cluster_id = cluster_id;
+                cfg.log_file = log_file;
+                cfg.enable_dynamic_config = true;
+                (cfg, v)
+            }
+            Err(e) => {
+                if let Some(PdError::Grpc(grpcio::Error::RpcFailure(status))) =
+                    e.downcast_ref::<PdError>()
```
Comment (overvenus) [inline cmd/src/server.rs]: Using `downcast_ref` is not recommended, you can define CfgError

```rust
enum CfgError {
  Other(Box<dyn error::Error + Sync + Send>),
  Pd(PdError)
}
```
Then we can use pattern match to get the umimplemented error.

**Comment by NingLin-P** [comment]
PTAL again @overvenus.

**Pull request merged**